src/ir/module-utils.h

```diff
@@ -56,8 +56,23 @@ struct BinaryIndexes {
       }
     };
     addIndexes(wasm.functions, functionIndexes);
-    addIndexes(wasm.globals, globalIndexes);
     addIndexes(wasm.events, eventIndexes);
+
+    Index globalCount = 0;
+    auto addGlobal = [&](auto* curr) {
+      globalIndexes[curr->name] = globalCount;
+      globalCount += curr->type.size();
```

Comment:
This code needs to map globals to their real indices in binaries. Tuple globals are split into multiple single-typed globals in the binary, so they need to occupy multiple indices.

Reply:
Oh, that sounds very confusing potentially :cry: 

How about not allowing tuple globals in binaries, and only in IR/wast files?

Reply:
This is the same thing that happens to tuple locals. Where else are global indices used where this could be confusing?

> How about not allowing tuple globals in binaries, and only in IR/wast files?

I think it would be surprising to have a valid wast that could not be assembled into a binary. It would also require changes to the test runner, which expects to be able to translate between the two.

Reply:
Maybe not in other places, good point, but it still seems confusing here.

I think this type of thing can be better done in a localized lowering pass. That is, instead of having some code in module-utils here and other elsewhere, we could have a `LegalizeGlobals` pass that converts such globals into bunches of normal ones. That seems simpler and more maintainable.

However, again, we'd only need to run such a pass if we actually want to write such globals. I think it would be fine to just error on that for now; we may never need it.

Reply:
`BinaryIndexes` is only used in wasm-binary.h and Print.cpp, but the global indices are only used in wasm-binary.h. So perhaps the global index calculation can be moved from module-utils to wasm-binary. Is that what you were thinking?

> However, again, we'd only need to run such a pass if we actually want to write such globals. I think it would be fine to just error on that for now; we may never need it.

If we want to validate tuple globals, then I would rather have testing and fuzzing for them as well to avoid users accidentally creating tuple globals via the API and being surprised when it doesn't work.

Reply:
If there's a way to localize this issue to just `wasm-binary.cpp` then that might work. I just want to not have to see or think about this in unrelated code.

Reply:
Talked offline and decided to just add a comment here explaining what is going on and to move `BinaryIndexes` out of module-utils in a follow-up PR.